state/store.go

```diff
@@ -414,6 +416,27 @@ func (store dbStore) SaveABCIResponses(height int64, abciResponses *tmstate.ABCI
 	return nil
 }
 
+// SaveBatchABCIResponseBytes saves a batch of ABCIResponses with given heights.
+func (store dbStore) SaveBatchABCIResponseBytes(heights []int64, abciResponseBytes [][]byte) error {
+	heightsLen := len(heights)
+	bytesLen := len(abciResponseBytes)
+	if heightsLen != bytesLen {
+		return fmt.Errorf("heights doesn't match abciResponseBytes: %d vs %d",
+			heightsLen,
+			bytesLen,
+		)
+	}
+	batch := store.db.NewBatch()
+	defer batch.Close()
+	for i, height := range heights {
+		err := batch.Set(calcABCIResponsesKey(height), abciResponseBytes[i])
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
```suggestion
		if err := batch.Set(calcABCIResponsesKey(height), abciResponseBytes[i]); err != nil {
			return err
		}
```
```